### DISCUSSION TITLE
Boolean return value?

### DISCUSSION THREAD

**Discussion by wcurti** [comfyanonymous/ComfyUI#1195] [Ideas]
If I want a node to return a value of True or False, what return type would I use? I don't want to have to convert to a STRING and then convert back, seems kludgy. 

It would also help to have an input type for a True/False flag. Is it just a COMBO?

**Comment by AitNev** [comment]
If you can use the Simple eval from the Efficiency Nodes extension. For your case you could return an int being 0 false and non 0 a true value then use the Evaluate Integers node with the expression a != 0 it can return an int or the string with the text True or Fale

**Comment by ltdrdata** [comment]
You can use 'BOOLEAN' type.
With the recent addition of booleans, there is no longer a need to use combo for booleans.

Just for your reference, you can configure the boolean widget as follows:
```
"mode": ("BOOLEAN", {"default": True, "label_on": "max", "label_off": "min"}),
```

**Reply by wcurti** [reply]
Ah, thanks. I think I need to do a pull.

**Reply by wcurti** [reply]
Where's the best way to find a list of all the return values?

**Reply by ntdviet** [reply]
> Where's the best way to find a list of all the return values?

Double click to open the search node dialog. On the right is the list of return values used by all the nodes you installed
![ảnh](https://github.com/comfyanonymous/ComfyUI/assets/54492570/01a90f2f-86e4-4299-a294-2e0b1420cd29)
